generated/php/google-cloud-speech-v1beta1/tests/unit/Speech/V1beta1/SpeechClientTest.php

```diff
@@ -293,4 +296,91 @@ public function asyncRecognizeExceptionTest()
         $this->assertTrue($grpcStub->isExhausted());
         $this->assertTrue($operationsStub->isExhausted());
     }
+
+    /**
+     * @test
+     */
+    public function streamingRecognizeTest()
+    {
+        $grpcStub = $this->createStub([$this, 'createMockSpeechImpl']);
+        $client = $this->createClient('createSpeechStubFunction', $grpcStub);
+
+        $this->assertTrue($grpcStub->isExhausted());
+
+        // Mock response
+        $resultIndex = 520358448;
+        $expectedResponse = new StreamingRecognizeResponse();
+        $expectedResponse->setResultIndex($resultIndex);
+        $grpcStub->addResponse($expectedResponse);
+        $grpcStub->addResponse($expectedResponse);
+        $grpcStub->addResponse($expectedResponse);
+
+        // Mock request
+        $request = new StreamingRecognizeRequest();
+
+        $bidi = $client->streamingRecognize();
+        $this->assertInstanceOf(BidiStream::class, $bidi);
+
+        $bidi->write($request);
+        $responses = [];
+        $responses[] = $bidi->read();
+
+        $bidi->writeAll([$request, $request]);
+        foreach ($bidi->closeWriteAndReadAll() as $response) {
+            $responses[] = $response;
+        }
+
+        $this->assertEquals([$expectedResponse, $expectedResponse, $expectedResponse], $responses);
```

Comment:
It is not that useful to compare those responses as they are all identical. (For example, if somehow the gapic layer messes around the ordering of the response, we would not be able to catch it here)